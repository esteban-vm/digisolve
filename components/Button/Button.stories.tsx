import type { Meta, StoryObj } from '@storybook/react'
import { screen, userEvent } from '@storybook/testing-library'
import Button from './Button'

export default {
  component: Button,
  title: 'Components/Button',
  parameters: { layout: 'centered' },
} as Meta<typeof Button>

export const ButtonExample: StoryObj<typeof Button> = {
  args: {
    children: 'Button Example',
    href: 'javascript:void(0)',
    isFull: true,
    role: 'button',
  },
  argTypes: {
    href: {
      control: false,
    },
    role: {
      control: false,
    },
    onClick: {
      action: true,
      control: false,
    },
  },
  play() {
    userEvent.click(screen.getByRole('button'))
  },
}
