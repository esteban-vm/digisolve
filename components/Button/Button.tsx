import { styled, css } from '@/styles'

type ButtonProps = {
  /** Is full or not */
  isFull?: boolean
}

const Button = styled.a<ButtonProps>`
  &:link,
  &:visited {
    text-decoration: none;
    padding: 10px 30px;
    border-radius: 200px;
    font-weight: 500;
    outline: none;
    transition: background-color 500ms;

    ${({ isFull }) => {
      if (isFull) {
        return css`
          background-color: var(--primary-color);
          color: #fff;

          &:hover,
          &:active {
            background-color: var(--primary-color-dark);
          }
        `
      }
    }};
  }
`

export default Button
