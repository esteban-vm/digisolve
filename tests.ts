import type { ReactElement } from 'react'
import { create, type ReactTestRendererJSON, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON() as ReactTestRendererJSON
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
