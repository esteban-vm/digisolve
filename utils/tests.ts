import type { ReactElement } from 'react'
import { create, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON()
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
