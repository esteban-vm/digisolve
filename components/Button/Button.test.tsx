import { render, screen, userEvent } from '@/tests'
import Button from './Button'

describe('Button test cases:', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Button href='#'>Test</Button>)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render full button', () => {
    const { asFragment } = render(
      <Button href='#' isFull>
        Test
      </Button>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it('should call click handler', async () => {
    const handleClick = jest.fn()

    render(
      <Button href='#' onClick={handleClick} role='button'>
        Test
      </Button>
    )

    await userEvent.click(screen.getByRole('button'))
    expect(handleClick).toHaveBeenCalled()
  })
})
