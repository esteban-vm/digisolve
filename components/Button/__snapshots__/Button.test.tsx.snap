// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test cases: should render correctly 1`] = `
<DocumentFragment>
  <a
    class="css-yhfbek"
    href="#"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Button test cases: should render full button 1`] = `
<DocumentFragment>
  <a
    class="css-1fps30p"
    href="#"
  >
    Test
  </a>
</DocumentFragment>
`;
